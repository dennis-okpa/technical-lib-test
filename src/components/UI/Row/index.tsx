import React from 'react'
import { Container } from './styles'
import Column from '../Column/index'

interface IRow {
  data: { [key: string]: string }[]
}

const Row: React.FC<IRow> = ({ data }) => {
  return (
    <Container>
      {data.map((columnData, index) => (
        <Column key={`grid-column-${index}`} data={columnData} />
      ))}
    </Container>
  )
}

export default Row
