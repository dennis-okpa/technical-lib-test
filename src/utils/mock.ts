export const mockLibraryData = {
  duration: 33.2277043024078,
  email: 'Ron.Hamill@gmail.com',
  firstName: 'Brooklyn',
  imageUrl:
    'https://images.pexels.com/photos/8422301/pexels-photo-8422301.jpeg',
  lastName: 'Flatley',
  likes: 88,
  thumbnailUrl:
    'https://images.pexels.com/photos/6994926/pexels-photo-6994926.jpeg?auto=compress&cs=tinysrgb&dpr=1&fit=crop&h=200&w=280',
}
