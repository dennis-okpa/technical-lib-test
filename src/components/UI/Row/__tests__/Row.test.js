import Row from '../index'
import React from 'react'
import { render } from '@testing-library/react'
import { mockLibraryData } from '../../../../utils/mock'

describe('Row', () => {
  it('should render correctly', () => {
    const { container } = render(<Row data={[mockLibraryData]} />)
    expect(container).toMatchSnapshot()
  })
})
