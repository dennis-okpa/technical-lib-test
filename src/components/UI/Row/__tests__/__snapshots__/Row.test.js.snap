// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should render correctly 1`] = `
<div>
  <div
    class="sc-bdvvtL jEqzPo"
  >
    <div
      class="sc-iCfMLu iTfSee"
    >
      <img
        alt="thumbnail"
        class="sc-furwcr biGNSl"
        src="https://images.pexels.com/photos/6994926/pexels-photo-6994926.jpeg?auto=compress&cs=tinysrgb&dpr=1&fit=crop&h=200&w=280"
      />
      <span
        class="sc-crHmcD coRCSh"
      >
        0:34
      </span>
      <div
        class="sc-pVTFL gnOtnb"
      >
        <span
          class="sc-jrQzAO eteWw"
        >
          Brooklyn
           
          Flatley
        </span>
        <span
          class="sc-kDTinF dbFsyY"
        >
          88
           
          <img
            class="sc-iqseJM gvxPJS"
            src="like.png"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
